Resolvable occurrenceID display

kyule opened the pull request:
for consideration on how it could look if we switch to a resolvable occurrenceID and add "IGSN:" to the list display  (it doesn't work until the occurrenceIDs are actually replaced locally though of course)

Note: this no longer names "SESAR" on the full record details, and it doesn't update the table display.

kyule committed:
occurrenceID display

for consideration on how it could look if we switch to a resolvable occurrenceID and add "IGSN:" to the list display  (it doesn't work until the occurrenceIDs are actually replaced)

Note: this no longer names "SESAR" on the full record details and it doesn't update the table display.

kyule committed:
undo gitignore change

kyule committed:
occurrenceIDs

-remove the "IGSN:" of the list search results to avoid confusion
- add "uuid:" when recordIDs are set, not I also updated the nightly maintenance stored procedure to copy recordID into the occurrenceID field
-set occurrenceID to the igsn uri
-attempt to update code for all necessary instances of occurrenceID  in OccurrenceSesar and IgsnManager

kyule commented:
@egbot @sunray1 I've made the changes that I think are necessary for updating the occurrenceIDs and recordIDs. However, I'm not that familiar with the IGSN/SESAR-related code (and it's not necessarily easy to test), so I definitely want both of you to review it when you can

sunray1 commented on collections/individual/index.php:
```diff
@@ -381,17 +381,29 @@ function initializeMap(){
+								<span style="margin-left: 10px"><?= $occArr['occurrenceid'] ?></span>
+								<span style="margin-left: 10px">
+									<a href="<?= $igsnUrl ?>" target="_blank">SESAR Record</a>
+								</span>
```
I think the link should be on the occurrence ID itself

kyule replied:
Totally works for me! I went back and forth on it.

sunray1 commented on collections/individual/index.php:
```diff
@@ -381,17 +381,29 @@ function initializeMap(){
+								</span>
+								<?php
+							}
+						}
```
I think catalognumber should be at the top

sunray1 commented on neon/classes/OccurrenceSesar.php:
```diff
@@ -1217,7 +1223,7 @@ public function generateIgsnSeed(){
 		if($this->collid){
 			$seed = 0;
-			$sql = 'SELECT MAX(occurrenceID) as maxid FROM omoccurrences WHERE occurrenceID LIKE "NEO%" AND length(occurrenceID) = 9';
+			$sql = 'SELECT MAX(occurrenceID) as maxid FROM omoccurrences WHERE occurrenceID LIKE "%NEO%" AND length(SUBSTRING_INDEX(o.occurrenceID, "igsn:10.58052/", -1)) = 9';
```
```suggestion
			$sql = 'SELECT MAX(occurrenceID) as maxid FROM omoccurrences WHERE occurrenceID LIKE "%NEO%" AND length(SUBSTRING_INDEX(occurrenceID, "igsn:10.58052/", -1)) = 9';
```

sunray1 committed:
Update OccurrenceSesar.php

sunray1 commented:
On reviewing this, I learned that SESAR added url verification to their sample registration - so you'll notice I just removed that field when in development mode

kyule committed:
Identifier links

Addresses Chandra's comments by making igsn id itself the link and listing catalog number first on the full record details pages.

kyule committed:
Fixes error in substring formula

Was previously missing the rest of the function needed to extract the igsn
